core/src/test/java/org/apache/calcite/util/graph/DirectedGraphTest.java

```diff
@@ -215,6 +215,35 @@
     assertEquals("[D, E]", frozenGraph.getShortestPath("D", "E").toString());
   }
 
+  @Test void testDistances() {
+    //       B -> C
```

Comment:
May I suggest another formatting to make it clearer that all edges are directed? e.g.:
```
// A+-->B+-->C+-->E
// +              ^
// |              |
// +----->D+------+
```

Reply:
Sounds reasonable. I have revised the code accordingly. Please check it.

Reply:
thanks, looks good.